client/post-editor/media-modal/index.jsx

```diff
@@ -160,6 +160,10 @@ export class EditorMediaModal extends Component {
 			filter: '',
 			detailSelectedIndex: 0,
 			source: props.source ? props.source : '',
+			dateRange: {
+				from: props.moment(),
```

Comment:
I think I used it after reading some best practise somewhere but I can't remember now.

I found this https://github.com/Automattic/i18n-calypso#moment-method

>>> The advantage of using a higher-order component instead of calling translate directly from the i18n-calypso module is that the latter does not properly account for change events which may be emitted by the state emitter object.

However, I _think_ we should be ok to switch to using it directly.

> My vote would be to import moment here if we can, or set the moment propType to be required.

I agree with you.

I think that having a default for `from` and `to` is a better experience. As long as you can override we are all good.

Reply:
> The advantage of using a higher-order component instead of calling translate directly from the i18n-calypso module is that the latter does not properly account for change events which may be emitted by the state emitter object.

Ahh ofc!

In that case, we could keep the dependency local by doing:

```js
import { localize } from 'i18n-calypso';

...
dateRange: { 
  from: props.moment(),
  to: props.moment(),
}
...

export default localize( EditorMediaModal );
```

As `moment` would then be supplied by the`localize()` HoC

Reply:
As discussed the default `export` returns the component wrapped in `localize` by default. Therefore the consumer is not required to pass in `moment`. The only time you need to pass in `moment` is if you deliberately access the non-wrapped component

```
import { DateRange } from 'date-range`
```

This is used in the tests where we stub out `moment` with a pre=configured instance so we can get some level of determinism in those tests (testing date stuff is hard for obvious reasons!).